designs/2022-community-eslint-org/README.md

```diff
@@ -0,0 +1,328 @@
+be seen as the "correct" place for popular packages.
+
+## Backwards Compatibility Analysis
+
+<!--
+    How does this change affect existing ESLint users? Will any behavior
+    change for them? If so, how are you going to minimize the disruption
+    to existing users?
+-->
+
+It won't affect existing ESLint users, as package could be released under the
+same name on `npm`.  
+Even when we would release a package under the `@eslint-community` npm scope
+(for instance because we can't get the rights to publish to the existing npm
+package), people could still use the old package without any extra problems
+(except for the ones they already had) if they wanted to.
+
+## Alternatives
+
+<!--
+    What other designs did you consider? Why did you decide against those?
+
+    This section should also include prior art, such as whether similar
+    projects have already implemented a similar feature.
+-->
+
+- Private persons could create a fork of the above mentioned packages.  
+  This could however create ceveral well maintained forks, which would cause
+  fragmentation in the community.
+
+  More importantly, this could cause the "problem" of the fork being reliant on
+  that specific individual, which could put us in the same position
+  (abandoned/unmaintained package) all over again at some point in time.
+  Possibly having a new fork each X amount of time, would also have the burden
+  of needing to update all existing dependants each time.
+
+- The community could independently create multiple more focused organizations
+  to maintain a subset of ESLint related packages.  
+  Some examples of already existing organizations like that are
+  [@import-js](https://github.com/import-js) (which maintains
+  [`eslint-plugin-import`](https://github.com/import-js/eslint-plugin-import)),
+  [@jsx-eslint](https://github.com/jsx-eslint) (which maintains
+  [`eslint-plugin-jsx-a11y`](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y),
+  [`eslint-plugin-react`](https://github.com/jsx-eslint/eslint-plugin-react) &
+  [`jsx-ast-utils`](https://github.com/jsx-eslint/jsx-ast-utils)) &
+  [@standard](https://github.com/standard) (which maintains a couple of
+  [`standard`](https://github.com/standard/standard) related ESLint configs)
+
+## Open Questions
+
+<!--
+    This section is optional, but is suggested for a first draft.
+
+    What parts of this proposal are you unclear about? What do you
+    need to know before you can finalize this RFC?
+
+    List the questions that you'd like reviewers to focus on. When
+    you've received the answers and updated the design to reflect them,
+    you can remove this section.
+-->
+
+- What exact acceptance criteria do we want to use?
+- What's the main place of communication for the "core" team?
+- Will there be a single team collectively maintaining all modules within the
+  project? Or will each module possibly have their own maintainers?
```

Comment:
> Or will each module possibly have their own maintainers?

👍 Having too many people with access is a security risk. A maintainer can invite a few collaborators(if want).

Reply:
> Having too many people with access is a security risk. a package maintainer can invite a few collaborators(if wanted).

What happens if that package maintainer disappears? When should the package maintainer be considered to have disappeared? After 3 months of inactivity? 6 months?

What would be considered "inactive"? Not answering issues / PR:s? Not keeping up to date with ESLint major releases? Not doing security releases?

When does the core team then step in? How do they then step in? By adding new co-maintainers? By merging / releasing stuff themselves? By replacing the package maintainer with a new one?

With individual maintainers for each module the organization doesn't give any redundancy in maintenance, it just ensures that a new maintainer can be put into place once the project is deemed abandoned. Hence its still likely that one ends up in the same situation as now. Only thing then fixed is that there's an easy process for putting a new maintainer in place. The wider ecosystem will still have been held back for X amount of time then though.

Reply:
> What happens if that package maintainer disappears? When should the package maintainer be considered to have disappeared? After 3 months of inactivity? 6 months?
> 
> What would be considered "inactive"?

I think that to disappear would be to become unresponsive via a private, personal channel.

Personally I think that a package maintainer should be contactable by "the committee" outside of GH (I.e. Email), and if they don't respond in some reasonable time frame (eg for NPM I think you've got 14 days to respond to a transfer request email) they're considered non-responsive, and "the committee" steps in to cleanup and appoint a new maintainer.

The beauty of some shared org is that the release process is centralised, so no need to  talk to NPM about an ownership transfer and no need to fork and republish under a new name. Just appoint new maintainers on GH and they can work to restore continuity of service - a process which would be invisible to users (I.e. Users might see a pause in updates but not have to make changes - the updates would just resume).